tests/contrib/test_multiprocess_file_storage.py

```diff
@@ -0,0 +1,293 @@
+
+_filehandle, FILENAME = tempfile.mkstemp('oauth2client_test.data')
+os.close(_filehandle)
+
+
+@contextlib.contextmanager
+def scoped_child_process(target, **kwargs):
+    die_event = multiprocessing.Event()
+    ready_event = multiprocessing.Event()
+    process = multiprocessing.Process(
+        target=target, args=(die_event, ready_event), kwargs=kwargs)
+    process.start()
+    try:
+        ready_event.wait()
+        yield
+    finally:
+        die_event.set()
+        process.join(5)
+
+
+def _create_test_credentials(expiration=None):
+    access_token = 'foo'
+    client_secret = 'cOuDdkfjxxnv+'
+    refresh_token = '1/0/a.df219fjls0'
+    token_expiry = expiration or (
+        datetime.datetime.utcnow() + datetime.timedelta(seconds=3600))
+    token_uri = 'https://www.google.com/accounts/o8/oauth2/token'
+    user_agent = 'refresh_checker/1.0'
+
+    credentials = OAuth2Credentials(
+        access_token, 'test-client-id', client_secret,
+        refresh_token, token_expiry, token_uri,
+        user_agent)
+    return credentials
+
+
+class MultiprocessStorageBehaviorTests(unittest2.TestCase):
+
+    def setUp(self):
+        try:
+            os.unlink(FILENAME)
+            os.unlink('{}.lock'.format(FILENAME))
+        except OSError:  # pragma: NO COVER
+            pass
+
+    def tearDown(self):
+        try:
+            os.unlink(FILENAME)
+            os.unlink('{}.lock'.format(FILENAME))
+        except OSError:  # pragma: NO COVER
+            pass
+
+    def test_basic_operations(self):
+        credentials = _create_test_credentials()
+
+        store = multiprocess_file_storage.MultiprocessFileStorage(
+            FILENAME, 'basic')
+
+        # Save credentials
+        store.put(credentials)
+        credentials = store.get()
+
+        self.assertNotEquals(None, credentials)
+        self.assertEquals('foo', credentials.access_token)
+
+        # Reset internal cache, ensure credentials were saved.
+        store._backend._credentials = {}
+        credentials = store.get()
+
+        self.assertNotEquals(None, credentials)
+        self.assertEquals('foo', credentials.access_token)
+
+        # Delete credentials
+        store.delete()
+        credentials = store.get()
+
+        self.assertEquals(None, credentials)
+
+    def _generate_token_response_http(self, new_token='new_token'):
+        token_response = json.dumps({
+            'access_token': new_token,
+            'expires_in': '3600',
+        })
+        http = HttpMockSequence([
+            ({'status': '200'}, token_response),
+        ])
+
+        return http
+
+    def test_single_process_refresh(self):
+        store = multiprocess_file_storage.MultiprocessFileStorage(
+            FILENAME, 'single-process')
+        credentials = _create_test_credentials()
+        credentials.set_store(store)
+
+        http = self._generate_token_response_http()
+        credentials.refresh(http)
+        assert credentials.access_token == 'new_token'
```

Comment:
You're in a `test_<something>` method in a `unittest2.TestCase` subclass; why isn't this `self.assertEqual('new_token', credentials.access_token)`?

Here and throughout this class.